app/models/miq_queue.rb

```diff
@@ -278,12 +278,31 @@ def self.put_or_update(find_options)
   #   puts a new item on the queue.  If the item was found, it will not be
   #   changed, and will be yielded to an optional block, generally for logging
   #   purposes.
+
+  # Return task.id
   def self.put_unless_exists(find_options)
-    put_or_update(find_options) do |msg, item_hash|
-      ret = yield(msg, item_hash) if block_given?
-      # create the record if the original message did not exist, don't change otherwise
-      ret if msg.nil?
+    find_options = default_get_options(find_options)
+    conds = find_options.dup
+    conds.delete(:state)
+    identifier = conds.hash.to_s
+
+    task = MiqTask.find_by(:identifier => identifier)
+
+    if task.nil?
+      task_id = MiqTask.create!(
+        :name       => 'put_unless_exists',
+        :state      => MiqTask::STATE_QUEUED,
+        :status     => MiqTask::STATUS_OK,
+        :identifier => identifier,
+        :message    => 'Queued'
+      ).id
+
+      put(conds.merge(:miq_callback => {:class_name => MiqTask.class.name, :instance_id => task_id, :method_name => :destroy}))
```

Comment:
you had said we needed to support legacy callbacks.
How many of these calls take a `miq_callback`?
Did we find only one?